packages/mobile/src/fiatExchanges/reducer.ts

```diff
@@ -31,19 +69,22 @@ export const reducer = (state: State = initialState, action: ActionTypes | Rehyd
     case Actions.SELECT_PROVIDER:
       return {
         ...state,
-        lastUsedProvider: {
-          name: action.name,
-          icon: action.icon,
-        },
+        lastUsedProvider: action.provider,
       }
     case Actions.ASSIGN_PROVIDER_TO_TX_HASH:
-      let providerToAssign = state.lastUsedProvider
-      if (!providerToAssign) {
+      // Don't override in case it came from Firebase (Actions.SET_PROVIDERS_FOR_TX_HASHES).
+      if (state.txHashToProvider[action.txHash]) {
+        return state
+      }
```

Comment:
I don't understand the comment. If action is `Actions.SET_PROVIDERS_FOR_TX_HASHES` why would it be handled here?

Reply:
I can see how the comment was confusing 😅 

We have two ways of assigning providers: The 'good way' that eventually will be the only way which is reading from Firebase (which is handled with the `SET_PROVIDERS_FOR_TX_HASHES` action) or assigning a tx hash that comes from a known partner address (the Bitrex ones) to the last provider clicked, which is of course much less accurate but it was the only way we had until now (and is still the only way for Simplex).

What I meant with the comment is that if a tx hash was already assigned using the 'good way' (through Firebase) we wouldn't want to override it with the error-prone temporary way.

I changed the comment to something more generic, let me know if you think I should be more specific with the comment.